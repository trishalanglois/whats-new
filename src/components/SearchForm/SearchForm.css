header {
  display: flex;
  justify-content: space-between;
  padding: 2em;
}

.search-form {
  grid-area: search;
}

input {
  font-size: 2em;
  background-color: #000;
  width: 70%;
  border-style: none;
}

#search-button {
  color: #fff;
  background-color: #000;
  font-size: 1.5em;
  margin-right: 1em;
  padding: .5em;
  border-radius: 50px 0px 0px 50px;
  text-align: right;
  width: 7em;
}

#search-button:hover,
  #search-button:focus {
    background-color: #A115F1;
    cursor: pointer;
  }
