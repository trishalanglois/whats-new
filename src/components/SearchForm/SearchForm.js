import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      keyword: ''
    }
  }


  handleChange = event => {
    this.setState({ keyword: event.target.value })
  }

  render () {
    return (
      <header>
        <form className = 'search-form'>
          <input type='search' placeholder='Search for a news article here' size='50' value={this.state.keyword} onChange={event => this.handleChange(event)} aria-label='find article based on keyword' />
        </form>
        <button id='search-button' onClick= {event =>
          this.props.filterArticles(event, this.state.keyword)
        }>Search Now</button>
      </header>
    )
  }
}

export default SearchForm
