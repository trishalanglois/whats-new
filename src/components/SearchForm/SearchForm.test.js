import React from 'react';
import SearchForm from './SearchForm';
import { shallow } from 'enzyme';

describe('SearchForm', () => {
  it('should correctly match the snapshot with all the data passed correctly', () => {
    const wrapper = shallow( < SearchForm
      filterArticles={jest.fn()}/>);
    expect(wrapper).toMatchSnapshot();
  })
})
