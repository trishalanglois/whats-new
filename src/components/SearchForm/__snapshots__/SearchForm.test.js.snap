// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should correctly match the snapshot with all the data passed correctly 1`] = `
<header>
  <form
    className="search-form"
  >
    <input
      aria-label="find article based on keyword"
      onChange={[Function]}
      placeholder="Search for a news article here"
      size="50"
      type="search"
      value=""
    />
  </form>
  <button
    id="search-button"
    onClick={[Function]}
  >
    Search Now
  </button>
</header>
`;
